pyomo/core/base/constraint.py

```diff
@@ -167,126 +168,145 @@ def __init__(self, expr=None, component=None):
         self._component = weakref_ref(component) if (component is not None) else None
         self._active = True
 
-        self._body = None
-        self._lower = None
-        self._upper = None
         self._expr = None
         if expr is not None:
             self.set_value(expr)
 
     def __call__(self, exception=True):
         """Compute the value of the body of this constraint."""
-        return value(self.body, exception=exception)
+        body = self.normalize_constraint()[1]
+        if body.__class__ not in native_numeric_types:
+            body = value(self.body, exception=exception)
+        return body
+
+    def normalize_constraint(self):
```

Comment:
I'm not sure I like the name of this method... 'Normalize' makes me think of means of 1 and stuff... Could it be `tupelize_constraint` or something?

Reply:
I'm not too keen on `tuplize_constraint` (it doesn't feel very descriptive); but I agree that `normalize_constraint` is probably worse.  What about `standardize_constraint` or `to_range_constraint` (the latter is a bit misleading as it does not return a `RangedExpression`)?

Reply:
Hmmm. I actually kind of like `to_range_constraint`... Maybe `to_tuple` (also not super descriptive, but since we accept Constraint expressions in that form, it kinda makes sense)? I don't love `standardize_constraint` because of potential standard form connotations, but I do like it a lot better than `normalize_constraint`!

Reply:
`to_bounded_expression()`?

Reply:
Oooo, I think I like that one!

Reply:
Implemented!